src/Mantasflowers.WebApi/Controllers/PaymentController.cs

```diff
@@ -16,19 +19,31 @@ namespace Mantasflowers.WebApi.Controllers
     public class PaymentController : ControllerBase
     {
         private readonly IPaymentService _paymentService;
+        private readonly IUserService _userService;
 
-        public PaymentController(IPaymentService paymentService)
+        public PaymentController(
+            IPaymentService paymentService,
+            IUserService userService
+            )
         {
             _paymentService = paymentService;
+            _userService = userService;
         }
 
         [AllowAnonymous]
         [HttpPost("create-checkout-session")]
         [ProducesResponseType(typeof(PostCreateCheckoutSessionResponse), StatusCodes.Status201Created)]
         [ProducesResponseType(typeof(string), StatusCodes.Status404NotFound)]
-        public async Task<IActionResult> CreateCheckoutSession(PostCreateCheckoutSessionRequest request)
+        public async Task<IActionResult> CreateCheckoutSessionAsync(PostCreateCheckoutSessionRequest request)
         {
-            var response = await _paymentService.CreateCheckoutSessionAsync(request);
+            Guid? userId = null;
+            try
+            {
+                string uid = User.GetUid();
+                userId = await _userService.GetUserGuidByUidAsync(uid);
+            } catch (Exception) { }
```

Comment:
Instead this exception catching, I thinkg you could just use `User.Identity.IsAuthenticated()` to check if user is authenticated. If he/she is, call the the methods in `Try {}`.

If `User.GetUid()` fails in such a case. `CreateCheckoutSessionAsync()` shouldn't be called because something is clearly wrong.